src/main/scala/ir/Statement.scala

```diff
@@ -25,6 +27,11 @@ class MemoryAssign(var lhs: Memory, var rhs: MemoryStore) extends Statement {
   override def acceptVisit(visitor: Visitor): Statement = visitor.visitMemoryAssign(this)
 }
 
+class NOP() extends Statement {
+  override def toString: String = "<NOP>"
+  override def acceptVisit(visitor: Visitor): Statement = this
+}
```

Comment:
Do we need this at all? Regardless, it should be an Object not a Class as there aren't distinct instances

Reply:
@yousifpatti

Reply:
It is just a placeholder (empty stmt). The data field could be made optional but that would require more refactoring. I agree with making it an object instead.